service/src/main/python/jcarbon/tensorflow/callbacks.py

```diff
@@ -125,6 +97,98 @@ def on_epoch_end(self, epoch, logs=None):
         self.reports[epoch] = self.last_report.to_frame().assign(
             epoch=epoch).set_index('epoch', append=True)
 
+class JCarbonExperimentCallback(JCarbonChunkingCallback):
```

Comment:
There are a number of nits I have here, but most of them are pitfalls in your code that you will almost certainly have to debug later.